android/app/src/main/java/updated/mysterium/vpn/ui/manual/connect/home/HomeActivity.kt

```diff
@@ -251,16 +258,18 @@ class HomeActivity : AppCompatActivity() {
     }
 
     private fun inflateNodeInfo() {
-        binding.nodeType.text = proposal.nodeType.getTypeLabel()
-        binding.nodeProvider.text = proposal.providerID
-        binding.pricePerHour.text = getString(
-            R.string.manual_connect_price_per_hour,
-            proposal.payment.rate.perSeconds / 3600.0
-        )
-        binding.pricePerGigabyte.text = getString(
-            R.string.manual_connect_price_per_gigabyte,
-            proposal.payment.rate.perBytes / 1024.0 / 1024.0
-        )
+        proposal?.let {
+            binding.nodeType.text = it.nodeType.getTypeLabel()
+            binding.nodeProvider.text = it.providerID
+            binding.pricePerHour.text = getString(
+                R.string.manual_connect_price_per_hour,
+                it.payment.rate.perSeconds / 3600.0
+            )
+            binding.pricePerGigabyte.text = getString(
+                R.string.manual_connect_price_per_gigabyte,
+                it.payment.rate.perBytes / 1024.0 / 1024.0
```

Comment:
Use constant vars for numbers for more context

Reply:
Fix